lib/cubit/payments/payments_cubit.dart

```diff
@@ -53,11 +53,17 @@ class PaymentsCubit extends Cubit<PaymentsState> with HydratedMixin {
 
   Future<PrepareSendResponse> prepareSendPayment({
     required String destination,
+    bool isDrain = false,
     BigInt? amountSat,
   }) async {
     _log.info("prepareSendPayment\nPreparing send payment for destination: $destination");
     try {
-      final req = PrepareSendRequest(destination: destination, amountSat: amountSat);
+      final payOnchainAmount = isDrain
+          ? const PayOnchainAmount_Drain()
+          : amountSat != null
+              ? PayOnchainAmount_Receiver(amountSat: amountSat)
```

Comment:
I'd prefer not to have **drain** related logic here at this stage as having partial implementation can make reviewing a bit difficult if/when we decide to integrate it in the UI.
```
final payOnchainAmount = (amountSat != null) ? PayOnchainAmount_Receiver(amountSat: amountSat) : null;
```
We can keep it as it is but let's add a TODO comment to state that the feature is not integrated in the UI yet.

---
  
Adding a getter would also make this more readable as nesting caused by ternary operator when nullable values are involved makes it harder to follow.
```
final payOnchainAmount = _getPayOnchainAmount(isDrain: isDrain, amountSat: amountSat);
```
```
PayOnchainAmount? _getPayOnchainAmount({requires bool isDrain, int? amountSat}) {
    if (isDrain) return const PayOnchainAmount_Drain();
    if (amountSat != null) return PayOnchainAmount_Receiver(amountSat: amountSat);
    return null;
}
```